### ISSUE TITLE
SC2185 doesn't account for flags

### ISSUE THREAD

**Issue by Thomazella** [koalaman/shellcheck#1312]
#### For bugs
- Rule Id:  SC2185
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
  find -Hx . -iname "*$1*"
```

#### Here's what shellcheck currently says:
```
Line 1	SC2185: Some finds don't have a default path. Specify '.' explicitly.
```
#### Here's what I wanted or expected to see:

Nothing, because I'm specifying `.` This is BSD find on MacOS. Man page says syntax is:
```
find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
````
It should account for those flags before the path.
